src/Ng/CommandHelpers.cs

```diff
@@ -34,6 +92,17 @@ static class CommandHelpers
             for (int i = 1; i < args.Length; i += 2)
             {
                 result.Add(args[i], args[i + 1]);
+                inputArgs.Add(args[i]);
+            }
+
+            var secretInjector = GetSecretInjector(result);
+
+            if (secretInjector != null)
```

Comment:
no need for this null check. you can have GetSecretInjector to return EmptySecretReader, that will just return the same string.